pkg/commands/compute/metadata_test.go

```diff
@@ -2,182 +2,174 @@ package compute_test
 	"github.com/fastly/cli/pkg/revision"
 	"github.com/fastly/cli/pkg/testutil"
 )
 
-// Scenario is an extension of the base TestScenario.
-// It includes manipulating stdin.
-type Scenario struct {
-	testutil.TestScenario
-
-	ExpectedConfig config.WasmMetadata
-}
-
 func TestMetadata(t *testing.T) {
-	var (
-		configPath string
-		data       []byte
-	)
-
-	// Create temp environment to run test code within.
-	{
-		wd, err := os.Getwd()
-		if err != nil {
-			t.Fatal(err)
-		}
-
-		// Read the test config.toml data
-		path, err := filepath.Abs(filepath.Join("./", "testdata", "metadata", "config.toml"))
-		if err != nil {
-			t.Fatal(err)
-		}
-		data, err = os.ReadFile(path)
-		if err != nil {
-			t.Fatal(err)
-		}
-
-		// Create a new test environment along with a test config.toml file.
-		rootdir := testutil.NewEnv(testutil.EnvOpts{
-			T: t,
-			Write: []testutil.FileIO{
-				{Src: string(data), Dst: "config.toml"},
-			},
-		})
-		configPath = filepath.Join(rootdir, "config.toml")
-		defer os.RemoveAll(rootdir)
-
-		if err := os.Chdir(rootdir); err != nil {
-			t.Fatal(err)
-		}
-		defer func() {
-			_ = os.Chdir(wd)
-		}()
+	// We read the static/embedded config so we can get the latest config
+	// version and so we don't accidentally switch to the UseStatic() version.
+	var staticConfig config.File
+	err := toml.Unmarshal(config.Static, &staticConfig)
+	if err != nil {
+		t.Error(err)
 	}
 
-	args := testutil.Args
-	scenarios := []Scenario{
+	scenarios := []testutil.TestScenario{
 		{
-			TestScenario: testutil.TestScenario{
-				Args:       args("compute metadata --enable"),
-				WantOutput: "SUCCESS: configuration updated",
+			Arg: "--enable",
+			ConfigFile: &config.File{
+				ConfigVersion: staticConfig.ConfigVersion,
+				CLI: config.CLI{
+					Version: revision.SemVer(revision.AppVersion),
+				},
 			},
-			ExpectedConfig: config.WasmMetadata{
-				BuildInfo:   "enable",
-				MachineInfo: "enable",
-				PackageInfo: "enable",
+			NewEnv: &testutil.NewEnvConfig{
+				EnvOpts: &testutil.EnvOpts{
+					Copy: []testutil.FileIO{
+						{
+							Src: filepath.Join("testdata", "metadata", "config.toml"),
+							Dst: "config.toml",
+						},
+					},
+				},
+				EditScenario: func(scenario *testutil.TestScenario, rootdir string) {
+					scenario.ConfigPath = filepath.Join(rootdir, "config.toml")
+				},
+			},
+			WantOutput: "SUCCESS: configuration updated",
+			Validator: func(t *testing.T, _ *testutil.TestScenario, opts *global.Data, _ error, _ bytes.Buffer) {
+				data, err := os.ReadFile(opts.ConfigPath)
+				if err != nil {
+					t.Error(err)
+				}
+
+				var testFile config.File
+				unmarshalErr := toml.Unmarshal(data, &testFile)
+				if unmarshalErr != nil {
+					t.Error(unmarshalErr)
+				}
+
+				testutil.AssertString(t, "enable", testFile.WasmMetadata.BuildInfo)
+				testutil.AssertString(t, "enable", testFile.WasmMetadata.MachineInfo)
+				testutil.AssertString(t, "enable", testFile.WasmMetadata.PackageInfo)
```

Comment:
@Integralist This code (and the other three copies of it) previously used `opts.Config.Read` to read the modified TOML file, and then validate that it had the desired configuration.

However, after refactoring the code into this new structure, that did not work: the values in the `opts.Config.WasmMetadata` struct were wrong for three of the four test cases. I added logging to display the actual contents of the file, and also directly unmarshaled it into another `config.File` object, and both showed the correct (expected) contents.

In the end I went with direct unmarshaling, since this test is really testing the `compute manifest` command's behavior and not the behavior of the `config` package.

Reply:
SGTM 👍🏻